flask_app.py

```diff
@@ -232,22 +254,75 @@ def check_identity_or_auth(request):
     if access_token is None:
         log.debug(f"Incoming request with no access token, telling them to auth the app")
         link = make_authorization_url()
-        resp = make_response(render_template('auth.html', authlink=link))
+        discordlink = make_discord_authorization_url()
+        resp = make_response(render_template('auth.html', authlink=link, discordauthlink=discordlink))
         return (resp, None)
 
     headers = {"Authorization": "bearer " + access_token, 'User-agent': 'CFB Risk Orders'}
     response = requests.get(REDDIT_ACCOUNT_URI, headers=headers)
     if response.status_code == 401:
-        log.error(f"{access_token},401 Error from CFBR API")
-        link = make_authorization_url()
-        resp = make_response(render_template('auth.html', authlink=link))
-        return (resp, None)
+        headers = {"Authorization": "Bearer " + access_token, 'User-agent': 'CFB Risk Orders'}
+        response = requests.get(DISCORD_ACCOUNT_URI, headers=headers)
+        if response.status_code == 401:
+            log.error(f"{access_token},401 Error from CFBR API")
+            link = make_authorization_url()
+            discordlink = make_discord_authorization_url()
+            resp = make_response(render_template('auth.html', authlink=link, discordauthlink=discordlink))
+            return (resp, None)
 
     # If we made it this far, we theoretically know the user's identity.  Say so.
+    
+    # First we provide whatever access token we got to reddit and see if it works? There's probably a better way to do this.
     username = get_username(access_token)
+
+    # If reddit doesn't know what's going on with that token, try Discord
+    if username == None:
+        # We add the $0 because that's what Mau is doing with discord verification right now.
+        # He wants to make it user ID eventually
+        username = get_discord_username(access_token)+"$0"
+
     return (None, username)
 
 
+###############################################################
+#
+# Discord API helper functions
+#
+###############################################################
+
+def make_discord_authorization_url():
+    params = {"client_id": DISCORD_CLIENT_ID,
+              "response_type": "code",
+              "redirect_uri": DISCORD_REDIRECT_URI,
+              "scope": "identify"}
+    url = f"{DISCORD_AUTH_URI}?{urllib.parse.urlencode(params)}"
+    return url
+
+def get_discord_token(code):
```

Comment:
This is so similar to `get_token()`, we could probably add a couple arguments and reduce the duplicate code.